sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -251,8 +254,15 @@ alloc get_pointer_type(const void *Ptr, const context &Ctxt) {
 
   // query type using PI function
   const detail::plugin &Plugin = CtxImpl->getPlugin();
-  Plugin.call<detail::PiApiKind::piextUSMGetMemAllocInfo>(
-      PICtx, Ptr, PI_MEM_ALLOC_TYPE, sizeof(pi_usm_type), &AllocTy, nullptr);
+  RT::PiResult Err =
+      Plugin.call_nocheck<detail::PiApiKind::piextUSMGetMemAllocInfo>(
+          PICtx, Ptr, PI_MEM_ALLOC_TYPE, sizeof(pi_usm_type), &AllocTy,
+          nullptr);
+
+  // PI_INVALID_VALUE means USM doesn't know about this ptr
```

Comment:
Shouldn't we handle `PI_INVALID_VALUE` error code separately?

Reply:
Not sure what you mean.

Reply:
Ah I messed up my default case.

Reply:
Reading this comment I would expect this function to return `alloc::unknown`, but I'm not sure what `piextUSMGetMemAllocInfo` does to `AllocTy` when it returns `PI_INVALID_VALUE` error code. It might leave it unmodified, then it's UB as `AllocTy` or it might set some value, which is not supposed to used by the caller because return code is not `PI_SUCCESS`.
I think it would be safe to add something like:
```c++
  // PI_INVALID_VALUE means USM doesn't know about this ptr
  if (Err == PI_INVALID_VALUE)
    return alloc::unknown;
  // Otherwise PI_SUCCESS is expected
  if (Err != PI_SUCCESS)
    throw runtime_error("Error querying USM pointer: ", Err);
```